wire-ios-sync-engine/Source/Use cases/SearchUsersUseCase.swift

```diff
@@ -0,0 +1,103 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireUtilities
+
+public protocol SearchUsersUseCaseProtocol {
+
+    func invoke(
+        query: String,
+        options: SearchOptions,
+        messageProtocol: MessageProtocol?
+    ) async throws -> SearchResult
+}
+
+public class SearchUsersUseCase: SearchUsersUseCaseProtocol {
+
+    // MARK: - Properties
+
+    private let context: NSManagedObjectContext
+    private let searchDirectory: SearchDirectory?
```

Comment:
Does this make sense to be optional? It seems like a critical part of the use case.

Reply:
No, it doesn't need to be optional. 
Changed